app/workers/rapid_ready_for_decision/form526_base_job.rb

```diff
@@ -26,7 +26,7 @@ def perform(form526_submission_id)
       begin
         with_tracking(self.class.name, form526_submission.saved_claim_id, form526_submission_id) do
           assessed_data = assess_data(form526_submission)
-          return if assessed_data.nil?
+          next if assessed_data.nil?
```

Comment:
This was a great opportunity to add a test, for checking that lack of data properly offramps the claim.

The test revealed that `return if assessed_data.nil?` would skip the after-yield code in `with_tracking`, thereby leaving the job in a `try` state rather than moving it over to `success`. In Ruby, `next` is the way to go if you just want to skip to the end of the block.

Reply:
Nice catch!

> leaving the job in a `try` state 

How did you determine the state of the job?

Reply:
[This line](https://github.com/department-of-veterans-affairs/vets-api/blob/d72cc85/spec/jobs/rapid_ready_for_decision/disability_compensation_job_spec.rb#L85) in the spec was failing with observed value `try` rather than `success`.